src/set.cr

```diff
@@ -152,6 +152,30 @@ struct Set(T)
     set
   end
 
+  # Difference: returns a new set containing elements in this set that are not
+  # present in the other.
+  #
+  #     Set.new([1,2,3,4,5]) - Set.new([2,4])               #=> Set{1, 3, 5}
+  #     Set.new(['a','b','b','z']) - Set.new(['a','b','c']) #=> Set{'z'}
+  def -(other : Set)
+    set = Set(T).new
+    each do |value|
+      set.add value unless other.includes?(value)
+    end
+    set
+  end
```

Comment:
I did think of that, particularly since Ruby allows that.  However, I wondered how often that is useful in practice.

In addition, performance characteristics of the operation are no longer predictable, because we don't know what `other` is.  It is better, in my opinion, to have some assurance of performance for all set operations.

Also, with an unrestricted instance of `Enumerable`, the exact type of the yielded item depends on the underlying object's type.  I am beginning to internalise, though, the fact that in Crystal, the compiler does eventually give an error when mismatched types are used.